### PR TITLE
fix(itly-2380): added browser field to sdk package.json for webpack 5

### LINKED ISSUE #111: Browser vs Node detection in @itly/sdk unreliable with Webpack 5
Webpack 5 removed the automatic polyfilling of Node globals like `process` in browser builds. However, it is common for users to add set the value of process.env.NODE_ENV and other variables using Webpack's DefinePlugin. This means that the lines below fail to detect the browser correctly and instead load the `node.ts` file, which is not compatible with the browser SDK. What was happening before was that it wasn't the `typeof process === 'undefined'`, `!p` check was causing the isBrowser boolean to be `true`, it was `p.browser === true` which is set by the [polyfill](https://github.com/defunctzombie/node-process/blob/master/browser.js#L156) (since removed).

https://github.com/amplitude/itly-sdk/blob/e7fab4c1ee0986bdc6bb6416de16a107da4254f2/packages/sdk/lib/index.ts#L43-L55

https://webpack.js.org/blog/2020-10-10-webpack-5-release/#automatic-nodejs-polyfills-removed

Possible solutions:
1. Set the `browser` field in package.json for the SDK to `dist/browser.js` so that bundlers like Webpack and Browserify know to use that instead of index.js. This would avoid feature detection altogether.
2. Use a more reliable way of detecting that the script is in a browser environment, such as checking for the `window` or `document`.

Workarounds for users:
1. Add the polyfill: https://github.com/Richienb/node-polyfill-webpack-plugin
2. Use DefinePlugin to set `'process.browser': true` in addition to `process.env` or anything else you might be defining.

### PR DESCRIPTION
**Pull request by justin-fiedler**
Fixes webpack 5 issues.

Closes #111

### CODE REVIEW

File: packages/plugin-iteratively/package.json [hunk 1]
```diff
@@ -34,11 +34,12 @@
     "url": "https://github.com/iterativelyhq/itly-sdk.git"
   },
   "dependencies": {
-    "@lukeed/uuid": "^2.0.0"
+    "uuid": "^8.3.2"
```
Comment (justin-fiedler) [inline packages/plugin-iteratively/package.json:37]: @falconandy thanks for updating this. It has been giving some users trouble. Unfortunately the new lib is much bigger (3.3kb vs 333b) and we are hoping to keep the SDK as light as possible for the browser. The [uuid README.md](https://www.npmjs.com/package/uuid) says it should play nice with tree shaking. Could you please check the size difference in the webpack output between the two? I we may not need this to be an uuid, so depending on the size difference we might want to find another solution. Thank you.

https://bundlephobia.com/package/@lukeed/uuid@2.0.0
https://bundlephobia.com/package/uuid@8.3.2

Comment (falconandy) [reply packages/plugin-iteratively/package.json:37]: @justin-fiedler 
I did tests on simple application that uses SDK, Schema Validator and Iteratively plugins. Webpack mode is `production`

Results:
| Branch | Webpack Version | Output Size (KB) |
|-|-|-|
| ITLY-2380-webpack-5-browser-detection | 5.41.0 | 52.5 |
| ITLY-2380-webpack-5-browser-detection | 4.46.0 | 54 |
| master | 5.41.0 + node-polyfill-webpack-plugin | 503 |
| master | 4.46.0 | 56 |

Detailed report for the `ITLY-2380-webpack-5-browser-detection` branch and Webpack 5.41.0 - [report.zip](https://github.com/amplitude/itly-sdk/files/6732779/report.zip)

File: packages/sdk/lib/browser.ts [hunk 2]
```diff
@@ -2,10 +2,42 @@
...
+  ValidationResponse,
 } from './base';
+import {
+  RequestLoggerPlugin,
+  RequestLogger,
+  ResponseLogger,
+} from './internal/RequestLogger';
+
+export {
+  Options,
```
Comment (justin-fiedler) [inline packages/sdk/lib/browser.ts:25]: @falconandy I think to make this work with codegen we need to export `ItlyBrowser` https://bitbucket.org/seasyd/analytics_tools/src/e3915adfd453ad616a305de14391102e8bc9f57c/sdk-examples/browser-javascript-v3/src/itly/index.js#lines-11

File: packages/plugin-iteratively/package.json (hunk 1 above)
Comment (justin-fiedler) [reply packages/plugin-iteratively/package.json:37]: This is great. Thank you.

**Pull request merged**